includes/admin/class-wc-rest-payments-reader-controller.php

```diff
@@ -62,4 +73,71 @@ public function get_summary( $request ) {
 		return rest_ensure_response( $summary );
 	}
 
+	/**
+	 * Proxies the get all readers request to the server.
+	 *
+	 * @param WP_REST_REQUEST $request Request object.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_all_readers( $request ) {
+		try {
+			return rest_ensure_response( $this->fetch_readers() );
+		} catch ( API_Exception $e ) {
+			return rest_ensure_response( new WP_Error( $e->get_error_code(), $e->getMessage() ) );
+		}
+	}
+
+	/**
+	 * Check if the reader status is active
+	 *
+	 * @param array  $readers The readers charges object.
+	 * @param string $id      The reader ID.
+	 * @return bool
+	 */
+	private function is_reader_active( $readers, $id ) {
```

Comment:
Not really needed, can be replaced with `$readers_ids =  array_column( $readers, 'reader_id' ); in_array( $id, $readers_ids, true)` right where the method is invoked.

Reply:
I have added one more check to that method and unfortunately, now it cannot be simplified.